rmgpy/molecule/molecule.py

```diff
@@ -551,6 +569,34 @@ def __reduce__(self):
         """
         return (Bond, (self.vertex1, self.vertex2, self.order))
 
+    def __hash__(self):
+        """
+        Define a custom hash method to allow Bond objects to be used in dictionaries and sets.
+        """
+        return hash(('Bond', self.order,
+                     self.atom1.symbol if self.atom1 is not None else '',
```

Comment:
I'm not sure, to be honest. It seems like it would make sense to have C-H and H-C return the same hash, but right now equality is defined as a memory id comparison, so this hash would provide acceptable behavior with fewer collisions. I'm open to changing it though.

Reply:
Let's keep it as is then, since it results in less collisions and we have memory id comparisons for equality checks.